src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1098,73 +1111,102 @@ private AmqpFrameHeaderFW setTransferFrame(
                 });
                 messageAnnotations = amqpMapRW.build();
             }
-            keyOffset.value = 0;
-            valueOffset.value = 0;
             return messageAnnotations;
         }
 
         private AmqpPropertiesFW properties(
-            ArrayFW<AmqpMessagePropertyFW> propertiesArray,
+            AmqpDataExFW dataEx,
             int bufferOffset)
         {
             AmqpPropertiesFW properties = null;
-            if (propertiesArray.fieldCount() > 0)
+            if (dataEx.properties().fieldCount() > 0)
```

Comment:
Please hoist out `dataEx.properties()` perhaps called `propertiesEx`.

Reply:
I kept using `dataEx.properties()` since there were name conflicts from `amqp.idl` and `protocol.idl` for `AmqpPropertiesFW`. I thought of using `AmqpTypeUtil` like other name conflicts but wasn't sure the best way to simplify this case.

Reply:
Can we change `protocol.idl` to use `AmqpMessageProperties` and `AmqpApplicationProperties`, leaving `AmqpProperties` for `amqp.idl`?